google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java

```diff
@@ -168,6 +169,23 @@
    */
   List<Key> allocateId(IncompleteKey... keys);
 
+  /**
+   * Reserve a list of IDs from a multiple complete key
+   *
+   * <p>Example of reserving multiple ids in a single batch.
+   *
+   * <pre>{@code
+   * KeyFactory keyFactory = datastore.newKeyFactory().setKind("MyKind");
+   * Key key1 = keyFactory.newKey(10);
+   * Key key2 = keyFactory.newKey("name");
+   * ReserveIdsResponse response = datastore.reserveIds(key1, key2);
+   *
+   * }</pre>
+   *
+   * @throws DatastoreException upon failure
+   */
+  ReserveIdsResponse reserveIds(Key... keys);
```

Comment:
Can we change the return type of this method? I don't think we want to pass the underlying proto object out of this method.  Instead I think we can return the list of reserved ids as a `List<Key>`.

Reply:
yes,we can explicitly return `List<Key>` because `reserveIds` APi return the empty response body,if its successful.

Reply:
Yeah, ReserveIdsResponse is empty and I thought it'd be more useful to users for us to take care of returning the ids that were reserved on success rather than them having to keep track of it out of band themselves. Thanks for updating.